### ISSUE TITLE
Support Amazon S3 Object Ownership

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#15471]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

> Amazon S3 Object Ownership is a new S3 feature that enables bucket owners to automatically assume ownership of objects that are uploaded to their buckets by other AWS Accounts. This helps you to standardize ownership of new objects in your bucket, and to share and manage access to these objects at scale via resource-based policies such as a bucket policy or an access point policy. Whether your S3 bucket receives data from other AWS accounts, or stores output from AWS services like AWS CloudTrail, S3 Object Ownership simplifies the work of creating and maintaining shared data sets on Amazon S3.  

One important note with this functionality: _S3 Object Ownership is currently under preview_ according to the AWS documentation, so that should also be noted in our documentation.

### New Resource

* aws_s3_bucket_ownership_controls

### Potential Terraform Configuration

```hcl
resource "aws_s3_bucket_ownership_controls" "example" {
  bucket = aws_s3_bucket.example.bucket

  rule {
    object_ownership = "BucketOwnerPreferred"
  }
}
```

### References

* [Announcement](https://aws.amazon.com/about-aws/whats-new/2020/10/amazon-s3-object-ownership-enables-bucket-owners-to-automatically-assume-ownership-of-objects-uploaded-to-their-buckets/)
* [S3 User Guide](https://docs.aws.amazon.com/AmazonS3/latest/dev/about-object-ownership.html)
* [AWS Go SDK service/s3](https://docs.aws.amazon.com/sdk-for-go/api/service/s3)

**Comment by shuheiktgw** [comment]
@bflad May I work on this issue if you haven't started it?

**Comment by cloudyparts** [comment]
This looks ready - will this be in the 3.10.0 release?

**Comment by stefansedich** [comment]
> This has been released in [version 3.10.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/v3.10.0/CHANGELOG.md). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
> 
> For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!
> 

Very cool! I have been waiting FOREVER for this feature, will be updating out buckets now!
